Source/WPF/MyMoney/Database/Money.cs

```diff
@@ -4246,6 +4271,48 @@ public string Symbol
             }
         }
 
+        [DataMember]
+        [ColumnMapping(ColumnName = "CultureCode", MaxLength = 80, AllowNulls=true)]
+        public string CultureCode
+        {
+            get { return this.cultureCode; }
+            set
+            {
+                if (this.cultureCode != value)
+                {
+                    
+                    this.cultureCode = Truncate(value, 80);
+                    if (this.cultureCode == "")
+                    {
+                        // Attempt to find a match using the Currency Symbol
+                        // this is not accuate since a currenty can be found in many countries, thus we pick the first match
+                        if (this.Symbol == "USD")
+                        {
+                            // Since we know the expected Culture for USD, lets use it instead of guessing. Since the first hit on the possible use of USD is "chr-US" == "Cherokee (United States)"
+                            this.cultureCode = "en-US";
+                        }
+                        else
+                        {
+                            // For all other code, we pick the first match
+                            var allCultureNames = CultureInfo.GetCultures(CultureTypes.SpecificCultures).Select(c => new { c, new RegionInfo(c.Name).ISOCurrencySymbol }).GroupBy(x => x.ISOCurrencySymbol).ToDictionary(g => g.Key, g => g.First().c, StringComparer.OrdinalIgnoreCase);

```

Comment:
Not sure if this algorithm is ever going to be useful, "en-USD" never matches "en-us" or "en-AUD" doesn't match "en-AU" and so on.  What if we simply create our own default mapping XML file from all known .NET cultures to all known currency symbols, we could even fill in the currency "name" by default from this lookup table, etc.

Reply:
See [list of culture codes](https://lonewolfonline.net/list-net-culture-country-codes/)

Reply:
I started with hardcoded mapping list, that but then I tried the this code and it does work. the input is just "USD" or "CAD" it will find a match, but its the first hit for that currency, for instance EUR has many countries that support it.  The good thing here is that the user can override this in the Currency-Table-View

Reply:
Just to confirm the input is not "en-USD", its just "USD" and it does match a few entries